src/apps/api/router.py

```
@@ -70,29 +118,191 @@ async def reindex_file(file_path: str):
+        # First update repository to latest (including submodules)
+        if not coordinator.git_manager.pull_changes():
+            raise HTTPException(status_code=500, detail="Failed to update repository")
+
+        import shutil
+        from pathlib import Path
+
+        db_path = coordinator.vector_store.persist_directory
+        if Path(db_path).exists():
+            try:
+                shutil.rmtree(db_path)
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)